trollflow2/tests/test_logging.py

```diff
@@ -76,36 +117,89 @@ def test_log_config_is_used_when_provided():
 
 def test_logging_works(caplog):
     """Test that the logs get out there."""
-    log = logging.getLogger("something")
-    with logging_on(log_queue):
-        logging.getLogger().addHandler(caplog.handler)
-        log.warning("oh no :(")
-        assert "oh no :(" in caplog.text
+    logger = logging.getLogger("something")
+    message = "oh no :("
+    with logging_on():
+        logger.warning(message)
+    assert message in caplog.text
 
 
-def fun(q, log_message):
-    """Fake a function to run."""
-    log = logging.getLogger('for fun')
-    setup_queued_logging(q)
-    log.debug(log_message)
-
-
-def run_subprocess(log_message, queue):
+def run_subprocess(loggers):
     """Run a subprocess."""
-    from multiprocessing import get_context
-    ctx = get_context('spawn')
-    proc = ctx.Process(target=fun, args=(queue, log_message))
+    proc = create_logged_process(target=fun, args=(loggers,))
     proc.start()
     proc.join()
 
 
+@queued_logging
+def fun(loggers):
+    """Fake a function to run."""
+    root_logger = logging.getLogger()
+    root_logger.debug("root debug")
+    root_logger.info("root info")
+    root_logger.warning("root warning")
+
+    for log_name in loggers:
+        logger = logging.getLogger(log_name)
+        logger.debug(f"{log_name} debug")
+        logger.info(f"{log_name} info")
+        logger.warning(f"{log_name} warning")
+
+
 @pytest.mark.skipif(sys.platform != "linux",
                     reason="Logging from a subprocess seems to work only on Linux")
-def test_logging_works_in_subprocess(caplog):
+def test_logging_works_in_subprocess_with_default_logging_config(caplog):
```

Comment:
This is actually the same test as `test_queued_logging_process_default_config`, I think it can be removed.

Reply:
I merged the two, `test_queued_logging_process_default_config` didn't have the duplicate lines check, so I added that.